navigation/src/main/kotlin/com/freeletics/khonshu/navigation/internal/NavEventCollector.kt

```diff
@@ -0,0 +1,51 @@
+package com.freeletics.khonshu.navigation.internal
+
+import com.freeletics.khonshu.navigation.NavRoot
+import com.freeletics.khonshu.navigation.NavRoute
+import kotlin.reflect.KClass
+
+
+public interface Navigator {
+    public fun navigateTo(route: NavRoute)
+    public fun navigateToRoot(root: NavRoot, restoreRootState: Boolean = false)
+    public fun navigateUp()
+    public fun navigateBack()
+    public fun <T : NavRoute> navigateBackTo(navRoute: KClass<T> , inclusive: Boolean = false)
+    public fun resetToRoot(root: NavRoot)
+}
+
+internal class NavEventCollector : Navigator {
```

Comment:
I think we can just merge these 2 into one class. The class can be public as long as we make the constructor and the `navEvents` property `internal`. That would also using a reified inline function for `navigateBackTo` like we do in `NavEventNavigator`